### ISSUE TITLE
Serverless V3 to V4 file provider regression

### ISSUE THREAD

**Issue by mhh611** [serverless/serverless#12935]
### Issue description

Hi,

I have several serverless files that use the `${file()}` resolver to resolve external files into the serverless file like this
```yaml
service: api
frameworkVersion: '4.x.x'

provider:
  name: aws
  runtime: nodejs18.x

resources:
  Resources:
    GraphQLAPI:
      Type: AWS::AppSync::GraphQLApi
      Properties:
        Name: api
        AuthenticationType: API_KEY
        AdditionalAuthenticationProviders:
          - AuthenticationType: AWS_IAM
        XrayEnabled: True

    GraphQLSchema:
      Type: AWS::AppSync::GraphQLSchema
      Properties:
        ApiId: !GetAtt GraphQLAPI.ApiId
        Definition: ${file(schema.graphql)}
```

with a schema.graphql file in the same directory
```graphql
type Channel {
  channelName: String! @aws_iam @aws_api_key
  data: String! @aws_iam @aws_api_key
}

type Mutation {
  publishMessage(channelName: String!, data: String!): Channel @aws_iam @aws_api_key
}

type Query {
  getChannel: Channel @aws_iam @aws_api_key
}

type Subscription {
  subscribe(channelName: String!): Channel
    @aws_subscribe(mutations: ["publishMessage"]) @aws_iam @aws_api_key
}
```

Using Serverless version `4.4.14` to deploy the file like `serverless deploy --region us-east-1` results in this erorr
```txt
✖ Error: Failed to resolve variable 'schema.graphql' with resolver 'file' and provider 'file': Error: Unsupported file type ".graphql" of file "/folder/schema.graphql"
    at #resolveKey (/.serverless/releases/4.4.14/package/dist/sf-core.js:693:51356)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async _ResolverManager.resolve (/.serverless/releases/4.4.14/package/dist/sf-core.js:693:56775)
    at async #handlePlaceholderNode (/.serverless/releases/4.4.14/package/dist/sf-core.js:693:55518)
    at async /.serverless/releases/4.4.14/package/dist/sf-core.js:693:53510
    at async processNodeAndHandleCompletion (/.serverless/releases/4.4.14/package/dist/sf-core.js:693:44152)
```

Serverless V4 appears to no longer support multiple file extensions such as `.graphql` or `.vtl` while it did in Serverless V3

### Context

Serverless ϟ Support

This generates a report from your last Serverless Framework command (including any errors) to use for Github Issues, debugging w/ AI, or creating a support ticket w/ Serverless Inc.

Summary Report -----------------

# Service Overview
- Serverless Framework Version: 4.4.14
- Service Config File: serverless.yml
- Service Name: api
- Service Runtime: nodejs18.x
- Command: deploy --region eu-west-1
- Error Code: RESOLVER_RESOLVE_VARIABLE_ERROR

## Service Path
/folder

## Error Message
Failed to resolve variable 'schema.graphql' with resolver 'file' and provider 'file': Error: Unsupported file type ".graphql" of file "/folder/schema.graphql"

## Error Stacktrace
```
at #resolveKey (file:///folder/.serverless/releases/4.4.14/package/dist/sf-core.js:693:51356)
at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
at async _ResolverManager.resolve (file:///folder/.serverless/releases/4.4.14/package/dist/sf-core.js:693:56775)
at async #handlePlaceholderNode (file:///folder/.serverless/releases/4.4.14/package/dist/sf-core.js:693:55518)
at async file:///folder/.serverless/releases/4.4.14/package/dist/sf-core.js:693:53510
at async processNodeAndHandleCompletion (file:///folder/.serverless/releases/4.4.14/package/dist/sf-core.js:693:44152)
```

## Service Config
```
service: api
frameworkVersion: '4.x.x'
provider:
  name: aws
  runtime: nodejs18.x
resources:
  Resources:
    GraphQLAPI:
      Type: AWS::AppSync::GraphQLApi
      Properties:
        Name: api
        AuthenticationType: API_KEY
        AdditionalAuthenticationProviders:
          - AuthenticationType: AWS_IAM
        XrayEnabled: True
    GraphQLSchema:
      Type: AWS::AppSync::GraphQLSchema
      Properties:
        ApiId: !GetAtt GraphQLAPI.ApiId
        Definition: ${file(schema.graphql)}
```

# Service Overview
- Serverless Framework Version: 4.4.14
- Command: support --summary true

Above is a summary report of your last Serverless Framework command.

Use it to debug with AI (e.g. ChatGPT) or report a new Github Issue: https://slss.io/issue

**Comment by czubocha** [comment]
Hey @mhh611, thanks for reporting. The fix will be included in the next release.

**Comment by czubocha** [comment]
Hi @mhh611, the fix has been included in [v4.4.15](https://github.com/serverless/serverless/releases/tag/v4.4.15). Feel free to try it out and let us know if you encounter any further issues or have additional feedback.
